### PR TITLE
Change argument of DoFRenumbering::downstream() to Tensor<1,dim>.

### PR DESCRIPTION
**Pull request by bangerth**
That's because this kind of argument indicates a *direction*, not a point, and
for the former Tensor<1,dim> is the correct data type.

### CODE REVIEW

**Comment by drwells** [comment]
The failing test appears to be related, so lets not merge this yet.

**Comment by bangerth** [comment]
Yep -- should have tested this even though I was working on the slow laptop :-( On it.

**Comment by bangerth** [comment]
That was subtle -- I had missed to adjust one function that was declared inline in the header file, and that now served both as an implementation and its corresponding declaration that continued to take a `Point` argument. I moved that function to the `.cc` file while there -- there is no need to inline such a function.

File: source/dofs/dof_renumbering.inst.in [hunk 1]
```diff
@@ -246,13 +246,19 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension :  SPACE_DIMENSIONS
      const Point<deal_II_dimension>&,
      const bool);
 
+    template
+    void downstream
+    (DoFHandler<deal_II_dimension>&,
```
Comment (tjhei) [inline source/dofs/dof_renumbering.inst.in]: I think you need instantiations for codim>0 and hp::DoFHandler (assuming that those work with the old inline function).

Comment (bangerth) [reply source/dofs/dof_renumbering.inst.in]: No, they never worked. `downstream` calls `compute_downstream`, which sits in the `.cc` file and was never instantiated for co-dimension DoFHandlers.

But I'm willing to do this in a follow-up, if you want.

Comment (tjhei) [reply source/dofs/dof_renumbering.inst.in]: > No, they never worked.

I am not surprised. ;-)

> But I'm willing to do this in a follow-up, if you want.

Not urgent, there is a pile of other things worth fixing before that.

**Comment by tjhei** [comment]
Can you fix the failing test?

**Comment by bangerth** [comment]
Let's try this one more time...

**Pull request merged**